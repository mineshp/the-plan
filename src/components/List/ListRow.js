import React from 'react';
import PropTypes from 'prop-types';
import { Button, Table, Icon, Label } from 'semantic-ui-react';

const projectLabels = (projects) => projects.map((project) => (
    <Label
        as="a"
        key={project.id}
        color="red"
    >
        <Icon name="empty star" />
        {project.name}
    </Label>
));

const ListRow = ({ data }) => {
    const listID = data._id; // eslint-disable-line no-underscore-dangle
    return (
        <Table.Row>
            <Table.Cell>{data.listName}</Table.Cell>
            <Table.Cell>{data.createdDate}</Table.Cell>
            <Table.Cell>{data.updatedDate}</Table.Cell>
            <Table.Cell>{projectLabels(data.projects)}</Table.Cell>
            <Table.Cell><Button as="a" href={`/list/${listID}/view`}>View</Button></Table.Cell>
        </Table.Row>
    );
};

ListRow.propTypes = {
    data: PropTypes.shape({
        _id: PropTypes.string.isRequired,
        listName: PropTypes.string.isRequired
    }).isRequired
};

export default ListRow;
