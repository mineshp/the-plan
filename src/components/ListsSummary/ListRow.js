import React from 'react';
import PropTypes from 'prop-types';
import { Button, Table, Label } from 'semantic-ui-react';
import { formatDate } from '../../helpers/validators/common';

const projectLabels = (projects) => projects.slice(0, 3).map((project) => (
    <Label
        as="a"
        key={project.id}
        color="purple"
        href={`/project/${project.name}/lists`}
    > {project.name}
    </Label>
));

const List = ({ data, onDeleteHandler, handleCompleted }) => {
    const listID = data._id; // eslint-disable-line no-underscore-dangle
    const rowProps = {};
    rowProps.negative = data.completed;

    const markListAsCompleteBtn = (
        <Button
            icon="thumbs up"
            color="orange"
            id={listID}
            onClick={handleCompleted}
            className="button-divider"
        />
    );

    const markListAsUnCompleteBtn = (
        <Button
            icon="check"
            color="green"
            id={listID}
            onClick={handleCompleted}
            className="button-divider"
        />
    );

    const completedBtn = data.completed ? markListAsUnCompleteBtn : markListAsCompleteBtn;

    return (
        <Table.Row {...rowProps}>
            <Table.Cell>{data.listName}</Table.Cell>
            <Table.Cell>{formatDate(data.createdDate)}</Table.Cell>
            <Table.Cell>{formatDate(data.updatedDate)}</Table.Cell>
            <Table.Cell>{projectLabels(data.projects)}</Table.Cell>
            <Table.Cell>
                {completedBtn}
                <Button
                    as="a"
                    color="blue"
                    icon="list layout"
                    href={`/list/view/${listID}`}
                />
                <Button
                    as="a"
                    icon="edit"
                    color="green"
                    href={`/list/update/${listID}`}
                />
                <Button
                    icon="trash"
                    color="pink"
                    value={listID}
                    onClick={onDeleteHandler}
                />
            </Table.Cell>
        </Table.Row>
    );
};

List.propTypes = {
    data: PropTypes.shape({
        _id: PropTypes.string.isRequired,
        listName: PropTypes.string.isRequired
    }).isRequired,
    onDeleteHandler: PropTypes.func.isRequired,
    handleCompleted: PropTypes.func.isRequired
};

export default List;
