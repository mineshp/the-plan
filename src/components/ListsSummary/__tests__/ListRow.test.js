import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Label } from 'semantic-ui-react';
import ListRow from '../ListRow';

const deleteList = jest.fn();

const mockListRow = {
    _id: '1234456',
    listName: 'My Awesome List',
    projects: [{
        id: '1',
        name: 'ABC'
    },
    {
        id: '2',
        name: 'XYZ'
    }]
};

describe('ListRow', () => {
    const mockListRowID = mockListRow._id; // eslint-disable-line no-underscore-dangle
    it('renders correctly', () => {
        const tree = renderer.create(<ListRow
            key={mockListRowID}
            data={mockListRow}
            onDeleteHandler={deleteList}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the correct number of project labels', () => {
        const wrapper = shallow(<ListRow
            key={mockListRowID}
            data={mockListRow}
            onDeleteHandler={deleteList}
        />);
        const ProjectLabels = wrapper.find(Label);

        expect(ProjectLabels.length).toBe(2);
    });
});
