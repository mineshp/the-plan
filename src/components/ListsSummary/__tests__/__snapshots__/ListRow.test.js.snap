// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRow marks the list row as complete if the list is marked as completed 1`] = `
<tr
  className="negative"
>
  <td
    className=""
  >
    Avengers - Age of Ultron
  </td>
  <td
    className=""
  >
    Invalid Date
  </td>
  <td
    className=""
  >
    Invalid Date
  </td>
  <td
    className=""
  >
    <a
      className="ui purple label"
      href="/project/Thor/lists"
      onClick={[Function]}
    >
       
      Thor
    </a>
    <a
      className="ui purple label"
      href="/project/Loki/lists"
      onClick={[Function]}
    >
       
      Loki
    </a>
  </td>
  <td
    className=""
  >
    <button
      className="ui green icon button button-divider"
      disabled={undefined}
      id="1234456"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="check icon"
      />
    </button>
    <a
      className="ui blue icon button"
      disabled={undefined}
      href="/list/view/1234456"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="list layout icon"
      />
    </a>
    <a
      className="ui green icon button"
      disabled={undefined}
      href="/list/update/1234456"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="edit icon"
      />
    </a>
    <button
      className="ui pink icon button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      value="1234456"
    >
      <i
        aria-hidden="true"
        className="trash icon"
      />
    </button>
  </td>
</tr>
`;

exports[`ListRow renders correctly 1`] = `
<tr
  className=""
>
  <td
    className=""
  >
    Avengers - Age of Ultron
  </td>
  <td
    className=""
  >
    Invalid Date
  </td>
  <td
    className=""
  >
    Invalid Date
  </td>
  <td
    className=""
  >
    <a
      className="ui purple label"
      href="/project/Thor/lists"
      onClick={[Function]}
    >
       
      Thor
    </a>
    <a
      className="ui purple label"
      href="/project/Loki/lists"
      onClick={[Function]}
    >
       
      Loki
    </a>
  </td>
  <td
    className=""
  >
    <button
      className="ui orange icon button button-divider"
      disabled={undefined}
      id="1234456"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
      />
    </button>
    <a
      className="ui blue icon button"
      disabled={undefined}
      href="/list/view/1234456"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="list layout icon"
      />
    </a>
    <a
      className="ui green icon button"
      disabled={undefined}
      href="/list/update/1234456"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="edit icon"
      />
    </a>
    <button
      className="ui pink icon button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      value="1234456"
    >
      <i
        aria-hidden="true"
        className="trash icon"
      />
    </button>
  </td>
</tr>
`;
